f5_openstack_agent/lbaasv2/drivers/bigip/tenants.py

```diff
@@ -72,12 +72,14 @@ def assure_tenant_created(self, service):
 
         # create tenant route domain
         if self.conf.use_namespaces:
-            for bigip in self.driver.get_all_bigips():
+            bigips = self.driver.get_all_bigips()
+            for bigip in bigips:
```

Comment:
I think my original idea is to pre-allocate the "next route domain id" before this loop. This implementation will still call create_route_domain() in this loop. Not sure if it can fix the problem either.

Reply:
Right, that would be still the problem as doing like this...